### ISSUE TITLE
Twisted's Sphinx documentation should allow linking to stdlib documentation.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4582]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Screwtape reported|
|-|-|
|Trac ID|trac#4582|
|Type|enhancement|
|Created|2010-07-13 13:21:05Z|
|Branch|https://github.com/twisted/twisted/tree/intersphinx-4582|

In ticket [#4567](https://github.com/twisted/twisted/issues/4567) comment 4, exarkun writes:

> I suppose eventually we can have a stdlib module role which is both fixed-width like code and a link. Does that make sense? Can you file a ticket for this? 

I don't know of any other Python projects whose docs link to the Python stdlib docs (especially since I don't think the Python stdlib has any version-independent permalinks), but it would be a nice feature.

<details><summary>Searchable metadata</summary>

```
trac-id__4582 4582
type__enhancement enhancement
reporter__Screwtape Screwtape
priority__normal normal
milestone__ 
branch__branches_intersphinx_4582 branches/intersphinx-4582
branch_author__julian julian
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1279027265000000 1279027265000000
changetime__1389976602000000 1389976602000000
version__None None
owner__Julian Julian
cc__khorn cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_1"></a>|@khorn commented|
|-|-|

This can be done one of two ways:

 1. Since the python stdlib docs use Sphinx, we can use Intersphinx, which allows links between Sphinx projects.  This has the advantage that it's a built-in Sphinx feature with no additional code to maintain.
 1. Create a custom extension with a stdlib role or something as exarkun suggested.  This would allow a bit more flexibility than the Intersphinx option as far as formatting, etc.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

+1 for intersphinx.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)|@khorn set owner to @khorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Replying to [khorn](#note_1):
> This can be done one of two ways:
> 
>  1. Since the python stdlib docs use Sphinx, we can use Intersphinx, which allows links between Sphinx projects.  This has the advantage that it's a built-in Sphinx feature with no additional code to maintain.

With InterSphinx you can use syntax like:

```
:py:mod:`urllib2`
```

in `conf.py`:

```python
# refer to the Python standard library.
intersphinx_mapping = {'python': ('http://docs.python.org', None)}
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_7"></a>|@khorn commented|
|-|-|

This should not be part of the Lore to Sphinx milestone

**Comment by twisted-trac** [comment]
|<img alt="julian's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|julian commented|
|-|-|

(In [41287]) Branching to intersphinx-4582.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_9"></a>|@Julian commented|
|-|-|

Here's a branch with the 5 lines to get this hooked up.

I've added two intersphinx sources, one for py2 and one for py3 -- intersphinx allows you to be explicit if we want to link to one or the other in places where it matters, otherwise it will just detect which one it needs to look at for unmabiguous refs.

If this is agreeable, I'll add a note somewhere on which prefixes we've got.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)|@khorn set owner to @khorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_11"></a>|@khorn set owner to @Julian|
|-|-|

Review notes:

 1. A note on which prefixes are set up, as you mentioned, would be nice.
 2. You didn't say anything about `intersphinx_cache_limit`.  I see no particular reason to change it from the default (5 days), but you might put a comment in the `conf.py` file so that it's obvious that this was a conscious decision, and not an oversight.

Don't block on the above though.  Please merge when ready.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_12"></a>|@khorn commented|
|-|-|

Forgot to mention, this probably also needs a topfile.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_13"></a>|@Julian set owner to @khorn|
|-|-|

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_14"></a>|@khorn set owner to @Julian|
|-|-|

Looks good.  Please merge.

**Comment by twisted-trac** [comment]
|<img alt="julian's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_15"></a>|julian set status to `closed`|
|-|-|

(In [41312]) Merge intersphinx-4582: Add support for intersphinx links to stdlib docs.

Author: Julian
Reviewers: khorn
Fixes: [#4582](https://github.com/twisted/twisted/issues/4582)

Adds py2 and py3 prefixes for the two corresponding stdlib documentation
indices.
